packages/accounts-controller/src/AccountsController.ts

```diff
@@ -429,6 +435,29 @@ export class AccountsController extends BaseController<
     });
   }
 
+  /**
+   * Updates the metadata of the account with the given ID.
+   *
+   * @param accountId - The ID of the account to set the name for.
+   * @param metadata - The new metadata for the account.
+   */
+  updateAccountMetadata(
+    accountId: string,
+    metadata: Partial<InternalAccount['metadata']>,
+  ): void {
+    const account = this.getAccountExpect(accountId);
+
+    this.update((currentState: Draft<AccountsControllerState>) => {
+      const internalAccount = {
+        ...account,
+        metadata: { ...account.metadata, ...metadata },
+      };
+      // Do not remove this comment - This error is flaky: Comment out or restore the `ts-expect-error` directive below as needed.
+      // // @ts-expect-error Known issue - `Json` causes recursive error in immer `Draft`/`WritableDraft` types
+      currentState.internalAccounts.accounts[accountId] = internalAccount;
```

Comment:
Since we already have this "non-error" popping out on `setAccountName` I wonder if we could refactor a bit `setAccountName` method and use `updateAccountMetadata` there too?

So this would be something like:
```ts
  setAccountName(accountId: string, accountName: string): void {
    if (
      this.listMultichainAccounts().find(
        (internalAccount) =>
          internalAccount.metadata.name === accountName &&
          internalAccount.id !== accountId,
      )
    ) {
      throw new Error('Account name already exists');
    }

    this.updateAccountMetadata(accountId, {
      name: accountName,
    });
  }
```

WDYT @danroc @montelaidev?